src/Blumchen/Subscriber/OptionsBuilder.Consumes.cs

```diff
@@ -0,0 +1,67 @@
+using System.Reflection;
+using System.Text.Json.Serialization;
+using Blumchen.Serialization;
+using Blumchen.Subscriptions.Replication;
+using JetBrains.Annotations;
+
+namespace Blumchen.Subscriber;
+
+public sealed partial class OptionsBuilder
+{
+    private INamingPolicy? _namingPolicy;
+    private JsonSerializerContext? _jsonSerializerContext;
+
+    [UsedImplicitly]
+    internal OptionsBuilder NamingPolicy(INamingPolicy namingPolicy)
+    {
+        _namingPolicy = namingPolicy;
+        return this;
+    }
+    
+    public interface INamingOptionsContext
+    {
+        OptionsBuilder AndNamingPolicy(INamingPolicy namingPolicy);
+    }
+
+    internal class NamingOptionsContext(OptionsBuilder builder): INamingOptionsContext
+    {
+        public OptionsBuilder AndNamingPolicy(INamingPolicy namingPolicy)
```

Comment:
**QUESTION:** Why do we need to define a naming policy for each contract? I think that having one per subscriber should be enough. Could you expand?

Reply:
That’s the intended usage. I’ll review that bit to enforce an at most once policy